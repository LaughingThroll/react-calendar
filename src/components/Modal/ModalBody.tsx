import React from "react"
import { TChildren } from "../../types/utilsTypes"

interface IModalBody {
  children?: TChildren
}

const ModalBody: React.FC<IModalBody> = ({ children }: IModalBody) => {
  return (
    <form className="modal-body">
      {children && !Array.isArray(children) && <div className="modal-body__item ">{children}</div>}
      {children &&
        Array.isArray(children) &&
        children.map((child, i) => (
          <div key={i} className="modal-body__item">
            {child}
          </div>
        ))}
    </form>
  )
}

export default ModalBody
