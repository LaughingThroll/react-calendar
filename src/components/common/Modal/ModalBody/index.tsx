import React from 'react'
import styles from './modalBody.module.scss'

interface ModalBodyProps {
  children?: any
  classNames?: string[]
}
const { modalBody } = styles

const ModalBody: React.FC<ModalBodyProps> = ({ children, classNames = [] }) => {
  return <div className={[modalBody, ...classNames].join(' ')}>{children}</div>
}

export default React.memo(ModalBody)
