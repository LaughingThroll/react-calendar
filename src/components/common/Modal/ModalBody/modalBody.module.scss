.modalBody {
  padding: 15px 27px 30px 38px;
}
