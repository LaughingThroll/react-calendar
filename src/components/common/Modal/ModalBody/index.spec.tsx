import { shallow, render } from 'enzyme'
import ModalBody from './'

describe('ModalBody render', () => {
  it('default render', () => {
    const tree = shallow(<ModalBody />)

    expect(tree).toMatchSnapshot()
  })

  it('render with children', () => {
    const tree = render(
      <ModalBody>
        {' '}
        <span></span> <span></span>{' '}
      </ModalBody>
    )

    expect(tree).toMatchSnapshot()
    expect(tree.find('span')).toHaveLength(2)
  })
})
