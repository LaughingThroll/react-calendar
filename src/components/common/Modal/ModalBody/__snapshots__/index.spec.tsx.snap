// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalBody render default render 1`] = `
<div
  className="modalBody"
/>
`;

exports[`ModalBody render render with children 1`] = `
<div
  class="modalBody"
>
   
  <span />
   
  <span />
   
</div>
`;
